src/black/__init__.py

```diff
@@ -749,11 +750,12 @@ def get_sources(
 
         # Compare the logic here to the logic in `gen_python_files`.
         if is_stdin or path.is_file():
-            root_relative_path = get_root_relative_path(path, root, report)
-
-            if root_relative_path is None:
+            if resolves_outside_root_or_cannot_stat(path, root, report):
```

Comment:
As you may have guessed from the name, the difference between this and `get_root_relative_path` (from #4161) is the `.resolve()`

Reply:
This helped me located where the filtering happens and I just added `sources.add(path)` before the old "continue"in line 755/756 and my problem was solved.
Tho this, if you need, I can still test it against my case where my home folder `~` was set to a symlink like `/homes/user` -> `/external/homes/user`. Please don't hesitate to tell me if you need any help! Thanks for the great work again!